dataset/src/main/scala/frameless/TypedDataset.scala

```diff
@@ -223,6 +223,26 @@ class TypedDataset[T] protected[frameless](val dataset: Dataset[T])(implicit val
   def take[F[_]](num: Int)(implicit F: SparkDelay[F]): F[Seq[T]] =
     F.delay(dataset.take(num))
 
+  /** Alias for [[firstOption()]].
+    */
+  def headOption[F[_]]()(implicit F: SparkDelay[F]): F[Option[T]] = firstOption()
+
+  /** Alias for [[take()]].
+    */
+  def head[F[_]](num: Int)(implicit F: SparkDelay[F]): F[Seq[T]] = take(num)
+
+  // $COVERAGE-OFF$
+  /** Alias for [[firstOption()]].
+    */
+  @deprecated("Use headOption or firstOption instead.", "0.5.0")
+  def head[F[_]]()(implicit F: SparkDelay[F]): F[Option[T]] = firstOption()
```

Comment:
I would match the vanilla signature to simplify migration. Deprecation is already harsh enough :smile:

Reply:
I though giving them method name was enough :D. Let me resolve the conflicts on this one after merging #153. I can make them just forward the result to vanilla head and first.